### ISSUE TITLE
refineBatchPFGS failed

### ISSUE THREAD

**Issue by AnthonyCoughlin** [OpenDroneMap/WebODM#101]
WebODM stuck in a loop.  Watching Docker the command line is still running but has been repeating the same 'warnings' and WebODM is throwing an error message 'refineBatchPFGS failed'.  See attached.

![dog_refinepatch_failed](https://cloud.githubusercontent.com/assets/25381671/22858251/00057002-f07d-11e6-8197-a2d3882a321c.JPG)

![webodm_bfgs_failed](https://cloud.githubusercontent.com/assets/25381671/22858252/06ac3a6c-f07d-11e6-8cee-e736c218ef02.JPG)

![webodm_bfgs_failed2](https://cloud.githubusercontent.com/assets/25381671/22858268/73b9e744-f07d-11e6-8d59-1ed1eeadfa0d.JPG)

Let me know if you need more info!

**Comment by AnthonyCoughlin** [comment]
Still going.

![webodm_bfgs_failed3](https://cloud.githubusercontent.com/assets/25381671/22858615/5be0e31a-f088-11e6-86f4-fb6eb2875bcd.JPG)

**Comment by pierotofy** [comment]
It is possible that the ReadTimeout error stopped execution. I'll have to investigate to tell for sure. refinePatchesBFGS often times fails during processing, but should not interrupt the process.

Restarting the task might fix it temporarily.

**Comment by AnthonyCoughlin** [comment]
@pierotofy  Thanks again for the assistance.  When this happened it corrupted the entire Docker db.  I've received db errors when trying to restart the Docker Machine.  I ended up wiping the WebODM folder again and uninstalling Docker Toolbox.  Then reinstalling Docker Toolbox and recloned WebODM.  I had to run `docker-compose build` to repair the db (not sure why I thought the uninstall would have removed all files) then ran `./webodm.sh update`.  Now WebODM starts with no errors.  However, I attempted to stitch the same dataset today (FYI - I uploaded the same data to MapsMadeEasy.com last night and it worked just fine - I'm pro ODM tho man!) and it failed.  I received this error (see below).  I am forcing focal-length and CCD width, I may be putting in the wrong values - would that cause an error?

![webodm_error_2-12-16](https://cloud.githubusercontent.com/assets/25381671/22866526/787f2792-f13d-11e6-9863-279547d87655.JPG)

**Comment by AnthonyCoughlin** [comment]
Also, watching the Docker cmd line, I keep seeing this warning (see below).  No sure if this helps you at all...

![webodm_background_wrapper](https://cloud.githubusercontent.com/assets/25381671/22866619/147eb904-f13f-11e6-9765-cc3a22e52d0e.JPG)

**Comment by pierotofy** [comment]
@AnthonyCoughlin thanks for the additional screens! No worries, I use mapsmadeeasy on occasion as well :)

It looks like the same problem is happening (ReadTimeout error). I will have time to look at this sometimes next week, I'll post an update with hopefully a fix then.

**Comment by AnthonyCoughlin** [comment]
@pierotofy, take your time.  Thanks again.

**Comment by AnthonyCoughlin** [comment]
@pierotofy,

Update:  I'm starting to believe there is something wrong with my dataset.  I successfully tested four other datasets yesterday evening, using both default settings and forced focal and CCD.  I'm planning on testing the failing dataset this evening and will get back with you.

**Comment by pierotofy** [comment]
I think the root cause of this problem lies in node-OpenDroneMap, and likely not on WebODM. Even though you received some error messages, the execution should have continued at the next cycle since the errors were thrown from a separate thread. Would you be able to share the dataset that you used?

**Comment by AnthonyCoughlin** [comment]
#104 

@pierotofy, Should I post or reference this issue/bug on the ODM git?  Whats the best way to share my images?

**Comment by pierotofy** [comment]
@AnthonyCoughlin Dropbox, or any other file sharing website that you like. If you can share the dataset publicly, just post the link here on Github. Otherwise send it here: https://www.masseranolabs.com/contact/.

I'll open an issue in the proper repository once I can reproduce it and confirm it.

**Comment by AnthonyCoughlin** [comment]
Hey @pierotofy,  Sorry for the delay. Sent you a link to images via the above mentioned link.  Thanks again for your help!

**Comment by pierotofy** [comment]
Got it.

I just ran the dataset and it processed correctly using the latest WebODM/node-OpenDroneMap.

![image](https://cloud.githubusercontent.com/assets/1951843/23094279/00369802-f5c4-11e6-9d2c-fb840ad81fc8.png)

I did use the code that was introduced in #103, so perhaps that could have made a difference, but I don't think it did. My guess is that perhaps your Docker containers don't have enough memory allocated, which is making the processing really, really slow.

Power down the Virtualbox "boot2docker" or "default" virtual machine from the VirtualBox Manager application, and follow these instructions to add more RAM: http://superuser.com/questions/926339/how-to-change-the-ram-allocated-to-an-os-in-virtualbox

Also, what happens if you add this processing node and use that instead? hostname: nodeodm.masseranolabs.com port: 80?
